src/transform/src/attribute/cardinality.rs

```diff
@@ -376,6 +376,129 @@ impl Factorizer for WorstCaseFactorizer {
     }
 }
 
+impl crate::analysis::Analysis for Cardinality {
+    type Value = SymExp;
+
+    fn announce_dependencies(builder: &mut crate::analysis::DerivedBuilder) {
+        builder.require(crate::analysis::Arity);
+        builder.require(crate::analysis::UniqueKeys);
```

Comment:
Shouldn't this also list `SubtreeSize`? If yes, then it's also surprising that no tests caught this.

Reply:
GitHub marked this as "Outdated" due to some code movements, but I think this still stands.

Reply:
Or is this a special Analysis that is always implicitly there? I can see it used in `ColumnNames` through `children_of_rev`, and it's also not announced there.

Reply:
> Or is this a special Analysis that is always implicitly there? 

That is correct. It's always required as part of the `DerivedBuilder::new(...)` call.

https://github.com/MaterializeInc/materialize/blob/5553085c2dff4d65ebd532828c7a185bfa6afd2f/src/transform/src/analysis.rs#L268-L279